087-external-certificate-manager.md

```diff
@@ -0,0 +1,294 @@
+5. Then on the next reconciliation, since the Kafka pods now have correct cert and key generation, copy over the new operator certificate.
+6. Since the Kafka pods now have correct cert generation, remove the old CA public cert from the `<CLUSTER_NAME>-cluster-ca-cert` Secret
+
+> ![Renewing the cluster CA public cert](./images/087-cert-renewals.png)
+>
+> Fig 2: Existing and proposed workflow when the user provides a new cluster CA public cert
+
+### User Operator/Clients CA
+
+When a new Kafka cluster is created Strimzi will copy the CA public cert from the Secret identified in `clientsCa.publicCert` to the `<CLUSTER_NAME>-clients-ca-cert` Secret.
+Strimzi will add two annotations to the `<CLUSTER_NAME>-cluster-ca-cert` Secret:
+* `strimzi.io/ca-cert-hash` with the value being a hash of the certificate.
+* `strimzi.io/ca-cert-generation` initially set to 0.
+
+#### Issuing KafkaUser certificates
+
+When a `KafkaUser` with `spec.authentication.type` set to `tls` is creates Strimzi  will create a `Certificate` custom resource.
+Strimzi will specify the required CN/SANS in the `Certificate` resource for the user certificate.
+Strimzi will specify the Secret for the certificate to be stored in as the existing name for the certificate Secret suffixed with `-cm`, for example `<USER_NAME>-cm`.
+Strimzi will request the certificate in both PEM and PKCS12 format.
+Strimzi will wait for the usual operation timeout during the reconciliation loop for the `Certificate` status to indicate that the certificate has been issued before continuing.
+Once the certificate has been issued, Strimzi will copy the certificate across from the cert-manager provided Secret into its own existing Secret.
+Strimzi will also copy across the ClientsCA public certificate.
+
+Strimzi will annotate the Secret it manages with:
+* `strimzi.io/user-cert-hash` annotation with the value being the hash of the certificate in the cert-manager Secret.
+* `strimzi.io/clients-ca-cert-generation` annotation with the value matching the current value on the `<CLUSTER_NAME>-clients-ca-cert` Secret.
+
+#### Tracking changes to user certificates
+
+Cert-manager will be responsible for renewing all user certificates.
+When a certificate is renewed cert-manager will update the related Secret.
+During a reconciliation Strimzi will check the hash of the certificate stored in cert-manager Secrets to see if an update is needed.
+Strimzi will copy over the new certificate into its own Secret and update the `strimzi.io/user-cert-hash` annotation to match the new certificate.
+
+> ![Issuing new user certificate](./images/087-new-renew-user-cert.png)
+>
+> Fig 3: Proposed workflow when cert-manager issues a new Kafka User certificate
+
+#### Handling CA cert renewals and key replacements
+
+During a reconciliation Strimzi will check the hash of the certificate stored in the user's CA public cert Secret to see if an update is needed.
```

Comment:
today we IMHO do not pass much information abotu the Clients CA to the UO and the UO instead uses Kube APi to fetch it from the Secret. But we might need to change it here as the UO will alos need to know what issuer to use etc.?